sage.py

```diff
@@ -1,49 +1,34 @@
-import os,re,operator, json, datetime, glob
-import statistics 
-import seaborn as sns
-import pandas as pd
-import requests
 import csv
+import datetime
+import glob
 import json
-import os.path
-import matplotlib.pyplot as plt 
-import itertools
-import numpy as np
-import matplotlib.pyplot as plt
-from numpy import diff
-from pandas import DataFrame
-import math
 import math
+import os
+import re
+import subprocess
+import sys
+from collections import defaultdict, OrderedDict
 from itertools import accumulate
+
 import matplotlib.pyplot as plt
-import matplotlib.style
-import matplotlib as mpl
-mpl.style.use('default')
 import numpy as np
-import seaborn as sns
-import numpy as np 
-import subprocess
-import sys
-import graphviz
-from shutil import copyfile
-import json
-import re
-from collections import defaultdict
+import requests
 
 # Import attack stages, mappings and alert signatures
 sys.path.insert(0, './signatures')
-from attack_stages import MicroAttackStage, MacroAttackStage
-from mappings import micro, micro_inv, macro, macro_inv, micro2macro, mcols, small_mapping, rev_smallmapping, verbose_micro, ser_groups
-from alert_signatures import usual_mapping, unknown_mapping, ccdc_combined, attack_stage_mapping
+from signatures.attack_stages import MicroAttackStage, MacroAttackStage
+from signatures.mappings import micro, micro_inv, macro, macro_inv, micro2macro, mcols, small_mapping, rev_smallmapping, verbose_micro, ser_groups
+from signatures.alert_signatures import usual_mapping, unknown_mapping, ccdc_combined, attack_stage_mapping
+
 
 IANA_CSV_FILE = "https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.csv"
-IANA_NUM_RETRIES = 3
+IANA_NUM_RETRIES = 5
 DB_PATH = "./ports.json"
 SAVE = True
 DOCKER = True
 
 
-def get_attack_stage_mapping(signature):
+def _get_attack_stage_mapping(signature):
```

Comment:
I have prefixed the helper methods with an underscore (so that the main ones are easier to find and differentiate)